addons/html_builder/static/src/core/utils.js

```diff
@@ -22,24 +22,26 @@ function isConnectedElement(el) {
     return el && el.isConnected && !!el.ownerDocument.defaultView;
 }
 
-export function useDomState(getState, { checkEditingElement = true, onReady } = {}) {
+export function useDomState(getState, { checkEditingElement = true } = {}) {
     const env = useEnv();
     const isValid = (el) => (!el && !checkEditingElement) || isConnectedElement(el);
-    const handler = () => {
+    const handler = async (ev) => {
         const editingElement = env.getEditingElement();
         if (isValid(editingElement)) {
-            Object.assign(state, getState(editingElement));
+            const newStatePromise = getState(editingElement);
+            ev?.detail.getStatePromises.push(newStatePromise);
+            const newState = await newStatePromise;
+            if (ev?.detail.updatePromise) {
+                const isLastBus = await ev?.detail.updatePromise;
```

Comment:
shouldApply ? or shouldIgnore ?

Reply:
It does not tell why it should apply or ignore so I'd keep `isLastBus` or `isLastBusMessage`. What do you think?

Reply:
Why he need to know the why ?